builds/misc/templates/base-image-publish/base-image-overview.yaml

```diff
@@ -0,0 +1,26 @@
+steps:
+- pwsh: |
+    # Note: With in the same job, you can access $(info), but it would interpret $(info) as string similar to `echo "$info"` which is undesirable for our case, we want an Object.
+    # We worked around this by using the serialization & deserailzation
+    $info = '$(serialize_info)' | ConvertFrom-Json
+
+    # Fetch tags & version from each module Dockerfiles {../$(arch)/base/Dockerfile, ../$(arch)/Dockerfile}
+    foreach ($each in $info)
+    {
+      $pathPrefix = "$(Build.SourcesDirectory)/$($each.Item1.ToString())/docker/$(os)/$(arch)"
+      $unberlyingFilePath = $pathPrefix + '/base/Dockerfile' | Resolve-path
+      $underlyingTag = $($(Get-Content $unberlyingFilePath | Select-String "ARG base_tag=") -split "=")[1]
+      $imageFilePath = $pathPrefix + '/Dockerfile' | Resolve-path
+      Set-Variable -Name "$($each.Item3.ToString())" -Value "$($($(Get-Content $imageFilePath | Select-String "ARG base_tag=") -split "=")[1])"
```

Comment:
I think we can disambiguate what Item3  and this value is by creating a variable for it with a label before this line. We could use this approach in other places as well.

Reply:
That would be a lot of duplication say... explicitly assigning {`edgeAgentTag`, `edgeHubTag`, `moduleTag`, `fullModuleTag`}. 

Note these variable are only set as env vars for this instance of `pwsh`, they are gone after this task is done.

Reply:
> That would be a lot of duplication say... explicitly assigning {edgeAgentTag, edgeHubTag, moduleTag, fullModuleTag}.

Yeah this is what I had in mind actually. It would have definitely helped me understand what this was doing better.

Reply:
Nope I'm not unrolling a loop with 5/6 lines repeating the same code over and over. 
What you request will turn this loop into this: 
```
      $pathPrefix = "$(Build.SourcesDirectory)/edge-agent/docker/$(os)/$(arch)"
      $unberlyingFilePath = $pathPrefix + '/base/Dockerfile' | Resolve-path
      $underlyingTag = $($(Get-Content $unberlyingFilePath | Select-String "ARG base_tag=") -split "=")[1]
      $imageFilePath = $pathPrefix + '/Dockerfile' | Resolve-path
      Set-Variable -Name "edgeAgentTag" -Value "$($($(Get-Content $imageFilePath | Select-String "ARG base_tag=") -split "=")[1])"
      Write-Output "azureiotedge-agent-base:$(Get-Variable -Name "edgeAgentTag" -ValueOnly) 🠘 $underlyingTag"


      $pathPrefix = "$(Build.SourcesDirectory)/edge-hub/docker/$(os)/$(arch)"
      $unberlyingFilePath = $pathPrefix + '/base/Dockerfile' | Resolve-path
      $underlyingTag = $($(Get-Content $unberlyingFilePath | Select-String "ARG base_tag=") -split "=")[1]
      $imageFilePath = $pathPrefix + '/Dockerfile' | Resolve-path
      Set-Variable -Name "edgeHubTag" -Value "$($($(Get-Content $imageFilePath | Select-String "ARG base_tag=") -split "=")[1])"
      Write-Output "azureiotedge-hub-base:$(Get-Variable -Name "edgeHubTag" -ValueOnly) 🠘 $underlyingTag"


      $pathPrefix = "$(Build.SourcesDirectory)/edge-modules/SimulatedTemperatureSensor/docker/$(os)/$(arch)"
      $unberlyingFilePath = $pathPrefix + '/base/Dockerfile' | Resolve-path
      $underlyingTag = $($(Get-Content $unberlyingFilePath | Select-String "ARG base_tag=") -split "=")[1]
      $imageFilePath = $pathPrefix + '/Dockerfile' | Resolve-path
      Set-Variable -Name "moduleTag" -Value "$($($(Get-Content $imageFilePath | Select-String "ARG base_tag=") -split "=")[1])"
      Write-Output "azureiotedge-module-base:$(Get-Variable -Name "moduleTag" -ValueOnly) 🠘 $underlyingTag"


      $pathPrefix = "$(Build.SourcesDirectory)/test/modules/TestAnalyzer/SimulatedTemperatureSensor/docker/$(os)/$(arch)"
      $unberlyingFilePath = $pathPrefix + '/base/Dockerfile' | Resolve-path
      $underlyingTag = $($(Get-Content $unberlyingFilePath | Select-String "ARG base_tag=") -split "=")[1]
      $imageFilePath = $pathPrefix + '/Dockerfile' | Resolve-path
      Set-Variable -Name "fullModuleTag" -Value "$($($(Get-Content $imageFilePath | Select-String "ARG base_tag=") -split "=")[1])"
      Write-Output "azureiotedge-module-base-full:$(Get-Variable -Name "fullModuleTag" -ValueOnly) 🠘 $underlyingTag"
```
I'd rather have a list somewhere and formulate the necessary variable from the list; rather than have multiple duplication of code everywhere you need it.

Reply:
Oh I didn't mean unrolling, I just meant adding labels to make it more readable.

Reply:
For example these lines took me a few minutes to understand:
```
      Set-Variable -Name "$($eachConfig.Item3.ToString())" -Value "$($($(Get-Content $imageFilePath | Select-String "ARG base_tag=") -split "=")[1])"
      Write-Output "$($eachConfig.Item2.ToString()):$(Get-Variable -Name "$($eachConfig.Item3.ToString())" -ValueOnly) 🠘 $underlyingTag"
```

I think having labels above these two lines for the values you are using within them would help a lot.